app/models/bookmark.rb

```diff
@@ -1,3 +1,10 @@
+class ControllerValidator < ActiveModel::EachValidator
+  def validate_each(record, attribute, value)
+    controllers = ["dashboard"] + (ActiveRecord::Base.connection.tables.map(&:to_s) + Permission.resources.map {|x| x.tableize }).uniq
```

Comment:
I think it's good enough, thanks.  The table list isn't going to change at all often, especially during runtime - so aggressively caching it is good.